### DISCUSSION TITLE
How to implement Algolia Search?

### DISCUSSION THREAD

**Discussion by FuadEfendi** [razonyang/hugo-theme-bootstrap#717] [Q&A] [answered]
Hello,

How to implement Algolia Search? I found example config in "params" file. I can design application which will create JSON objects from existing Markdown files and upload it to Algolia, but I can't understand yet how to link search results to website public URLs, especially using anchors (maybe each anchored snippet will be separate JSON object). It seems HBS website uses just free "submit your URL" version of Algolia which will then crawl website?

Thanks,

**Comment by razonyang** [comment, marked as answer]
**EDITED**

Algolia is supported now, see also https://hbs.razonyang.com/v1/en/docs/configuration/site-params/.

Preview online on https://razonyang.com/.

![image](https://user-images.githubusercontent.com/17720932/204083278-a925fc8d-ea73-400e-b598-ec547011ea27.png)


<hr>

Hmm, I couldn't provide helpful information for you. The theme supports [DocSearch](https://docsearch.algolia.com/) which is powered by Algolia, but does not support Algolia. You may need to implement it from scratch.

You can also apply DocSearch if your site meets the [requirements](https://docsearch.algolia.com/docs/who-can-apply/#the-checklist). Once the request approved, you'll receive the required information to [configure](https://hbs.razonyang.com/v1/en/docs/configuration/site-params/) your site, the parameters starts with `docsearch`.

**Reply by FuadEfendi** [reply]
Probably [docsearch] is what exactly I need... do you upload manually JSON to Algolia, or it is done automatically?  

My `Fuse.js` raw index size it 11Mb (and compressed: 1Mb); and it is only 100 characters from article headers, no any "synonym dictionary" support, word forms support, etc; very basic; and I don't like its' current UI: it highlights individual characters. I am exprrt on Lucene/Solr... first experience with Fuse.js and it is super simple & attractive for BLOG-sites.

**Reply by FuadEfendi** [reply]
Sorry I didn't read in-full; yes, I need to implement JSON-upload from scratch, DocSearch is limited to free open-source documentation sites only, and I don't want their crawler to crawl 25000 pages from my site (including pagination, tags, "similar terms", etc.) - and they don't want neither.

**Comment by razonyang** [comment]
Hi @FuadEfendi, the theme supports Algolia (basic functionalities) now, see https://hbs.razonyang.com/v1/en/docs/configuration/site-params/ for details if you're still interested in it.
